### DISCUSSION TITLE
function call 多个工具合并成一个

### DISCUSSION THREAD

**Discussion by GXKIM** [zai-org/ChatGLM3#236] [General]
如题，有什么建议吗？

**Comment by zRzRzRzRzRzRzR** [comment]
您是想一次调用多个工具吗

**Comment by GXKIM** [comment]
> 您是想一次调用多个工具吗

工具1：机器学习相关的书籍名称，工具2：根据返回的书籍名称，也就是名称，去调用数据库工具，把这俩合并成一个

**Reply by zRzRzRzRzRzRzR** [reply]
我们的建议是：
1. 如果你这两个工具是连续的，也就是工具一结束后直接调用工具二，则直接把1 2 两个工具写在一个新工具中
2. 如果完成2 的前提不是完成工具1 那你应该用顺序执行两个工具，或者做好接口只执行工具2 ，而不是将其合并在一个工具
模型目前仅支持顺序调用工具，不具备同时调用多个工具

**Reply by GXKIM** [reply]
感谢建议，想请问作者大佬，定义工具的时候，比如我定义为，‘回答所有机器学习相关的问题的tool，源码中举的例子都是有retunrn值，这种我应该怎么给return值呢([ChatGLM3/tool_register.py)’

**Reply by zRzRzRzRzRzRzR** [reply]
你的return值不一定是一个str，你可以是一个结果，一个字典都是可以的

**Reply by GXKIM** [reply]
感谢大佬，已经实现
